core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2117,4 +2117,15 @@ package object config {
       // batch of block will be loaded in memory with memory mapping, which has higher overhead
       // with small MB sized chunk of data.
       .createWithDefaultString("3m")
+
+  private[spark] val MARK_FILE_LOST_ON_EXECUTOR_LOST =
+    ConfigBuilder("spark.shuffle.markFileLostOnExecutorLost")
```

Comment:
I am having similar concern with @viirya and @attilapiros . I think we should not make it as a user-facing config. If we would like introducing a config for this anyway, it'd better to start with `internal()` config, but not user-facing.

Though `spark.shuffle.manager` can be arbitrary class, but in practice, they are just a handful of implementation in one company's environment, and ideally the infra developers should control which implementation to use, instead of user to control this. Similarly for whether to mark shuffle file lost should be controlled by developers team but not users.

Just share some context, in FB, we (spark developer) control these kinds of behavior transparently and make this invisible to spark user. This also helps us to migrate to newer implementation easier without worrying about users setting wrong config. The mixed case (query uses customized shuffle service and default shuffle service) can happen quite a bit in production, as we have rate limit for traffic on customized service, and need fallback.

Reply:
Hi Cheng, thanks for the comment, and known you from FB! We previously discussed with people from FB about Cosco, but not for this specific issue. Just curious, do you guys modify your internal Spark distribution to make executor lost event not trigger driver to delete map output?

Reply:
> do you guys modify your internal Spark distribution to make executor lost event not trigger driver to delete map output?

Yeah I think so, I can do a double check with cosco folks as well tomorrow. btw nice to know you as well!

Reply:
Cool, that will be interested to know, thanks!

Reply:
@c21 May I ask what is the status of open sourcing Cosco? 
I just think it might help us to make better decisions about the Shuffle plugin API of Spark if we could look into that one too.

Reply:
@attilapiros - unfortunately there's no specific ETA now, but the team is working on it.